core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -45,26 +53,30 @@
    *
    * @param logger the logger to use for this call
    * @param callable the callable to call
-   * @param logAnyFailure whether to log whenever the RPC fails
+   * @param logAnyFailure whether to log or increment metrics whenever the RPC fails
    * @param <T> the return type of the callable
    * @return the return value from calling the callable
    * @throws AlluxioTException if the callable throws an exception
    */
   public static <T> T call(Logger logger, RpcCallable<T> callable, boolean logAnyFailure)
       throws AlluxioTException {
-    try {
+    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(callable)).time()) {
       logger.debug("Enter: {}", callable);
       T ret = callable.call();
       logger.debug("Exit (OK): {}", callable);
       return ret;
     } catch (AlluxioException e) {
       logger.debug("Exit (Error): {}", callable, e);
-      if (logAnyFailure && !logger.isDebugEnabled()) {
-        logger.warn("{}, Error={}", callable, e.getMessage());
+      if (logAnyFailure) {
+        MetricsSystem.counter(getQualifiedFailureMetricName(callable)).inc();
```

Comment:
why this inc is guarded by `logAnyFailure` whereas line 79 is not?

Reply:
If `logAnyFailure` is true, it means any `AlluxioException` is considered a real failure, so we increment the counter. on line 79, a `RuntimeException` is caught which is always a failure, so we should always increment the counter.

Reply:
can you javadoc this reasoning? It is not obvious to me

Reply:
Updated, thanks.